src/hnsw/build.c

```diff
@@ -221,28 +222,42 @@ static void ScanTable(HnswBuildState *buildstate)
 static void BuildIndex(
     Relation heap, Relation index, IndexInfo *indexInfo, HnswBuildState *buildstate, ForkNumber forkNum)
 {
+    InitBuildState(buildstate, heap, index, indexInfo);
+
+    usearch_error_t        error = NULL;
     usearch_init_options_t opts;
     MemSet(&opts, 0, sizeof(opts));
-    InitBuildState(buildstate, heap, index, indexInfo);
 
     opts.dimensions = buildstate->dimensions;
     PopulateUsearchOpts(index, &opts);
 
-    usearch_error_t error = NULL;
-    buildstate->hnsw = NULL;
     buildstate->usearch_index = usearch_init(&opts, &error);
-
     elog(INFO, "done init usearch index");
     assert(error == NULL);
 
-    usearch_reserve(buildstate->usearch_index, 1100000, &error);
-    assert(error == NULL);
+    buildstate->hnsw = NULL;
+    if(buildstate->index_file_path) {
+        usearch_load(buildstate->usearch_index, buildstate->index_file_path, &error);
+        assert(error == NULL);
+        elog(INFO, "done loading usearch index");
+
+        // todo determine how we can set index rd_options
+        // HnswOptions *opts = (HnswOptions *)index->rd_options;
+        // opts->m = usearch_connectivity(buildstate->usearch_index, &error);
+        // assert(error == NULL);
+        // opts->dims = usearch_dimensions(buildstate->usearch_index, &error);
```

Comment:
You can probably add all these (usearch_connectivitydims ) to existing [`usearch_metadata`](https://github.com/Ngalstyan4/usearch/blob/c9454d98e8af2402a1bd874db182dd8530d44350/c/lib.cpp#L172-L181)

Reply:
@var77 I assume you left these comments in to continue working on these in a separate PR?

Reply:
Yes I will do the required changes on a separate PR and remove this comments once done.